src/file.cpp

```diff
@@ -1330,6 +1351,7 @@ namespace libtorrent
 	bool file::open(std::string const& path, int mode, error_code& ec)
 	{
 		close();
+		native_path_string const &file_path = convert_to_native_path_string(path);
```

Comment:
how can this function return a reference to the string? where is the storage for it?